### PR TITLE
Remove the S3KeyPairProvider

### PR DESCRIPTION
**Pull request by yjpa7145**

### CODE REVIEW

File: packages/api/lib/cwSfExecutionEventUtils.js [hunk 1]
```diff
@@ -23,6 +19,15 @@ const executionStatusToWorkflowStatus = (executionStatus) => {
   return statusMap[executionStatus];
 };
 
+const getTaskExitedEventOutput = (event) =>
```
Comment (Jkovarik) [inline packages/api/lib/cwSfExecutionEventUtils.js:22]: Moving these here seems right, but there's still a unit test here:

https://github.com/nasa/cumulus/blob/CUMULUS-1684-remove-S3KeyPairProvider-2/packages/common/tests/test-execution-history.js


Why is this passing in CI?     Caching?

Comment (Jkovarik) [reply packages/api/lib/cwSfExecutionEventUtils.js:22]: Anyway, I'd expect [`6f5a252` (#1800)](https://github.com/nasa/cumulus/pull/1800/commits/6f5a2525d2f10fbd4c29d47ede3e8cd07dea521e) should fix it.

File: packages/common/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
   },
   "scripts": {
     "build-docs": "npm run tsc && ../../node_modules/.bin/jsdoc2md --template templates/docs-API.hbs URLUtils.js util.js > docs/API.md",
-    "clean": "rm -f *.d.ts *.map",
+    "clean": "rm -f *.d.ts *.map && ls *.js | grep -v nyc.config.js | xargs rm -f",
```
Comment (Jkovarik) [inline packages/common/package.json:27]: 👍    We're going to need to be mindful of this behavior if adopted as a pattern, as there are <other things?> that use .js configs out there.

File: tf-modules/archive/main.tf [hunk 3]
```diff
@@ -10,3 +10,9 @@ locals {
     var.elasticsearch_security_group_id
   ])
 }
+
+data "aws_lambda_invocation" "verify_provider_secrets_migration" {
```
Comment (Jkovarik) [inline tf-modules/archive/main.tf:14]: So this activates the verification lambda.... but that lambda has the following in it: 


```
 else {
    if (isNil(provider.username) && isNil(provider.password)) return;
    throw new Error(`Provider ${provider.id} has plaintext username or password. Must invoke the providerSecretsMigration Lambda function.`);
  }
```

We should probably update that logic, given we're removing the lambda and that should no longer be possible.

File: packages/api/lib/cwSfExecutionEventUtils.js (hunk 1 above)
Comment (yjpa7145) [reply packages/api/lib/cwSfExecutionEventUtils.js:23]: Yeah, that was my mistake. Fixed now.

File: packages/common/package.json (hunk 2 above)
Comment (yjpa7145) [reply packages/common/package.json]: This should be safer: 380445e91c8afe5c143440b2620e3b747dea16e8

**Comment by Jkovarik** [comment]
380445e - nice.   👍

**Pull request closed without merging**